static/js/apps/eval_rig/data_store.tsx

```diff
@@ -34,10 +35,48 @@ import {
   DC_STAT_FEEDBACK_COL,
   LLM_STAT_FEEDBACK_COL,
   QUERY_ID_COL,
+  QUERY_OVERALL_FEEDBACK_COL,
   USER_COL,
 } from "./constants";
 import { Response } from "./types";
 
+// Gets the document path for a specific query or call
+export function getPath(
+  sheetId: string,
+  queryId?: number,
+  callId?: number
+): string[] {
+  const path = ["sheets", sheetId];
+  if (queryId) {
+    path.push(...["queries", String(queryId)]);
+  }
+  if (callId) {
+    path.push(...["calls", String(callId)]);
+  }
+  return path;
+}
+
+// Sets a field in a doc at the specified path
+export async function setField(
+  path: string[],
+  fieldKey: string,
+  fieldValue: string
+): Promise<void> {
+  const docRef = doc(db, path.shift(), ...path);
```

Comment:
why need to do path.shift() here?

Reply:
ah, originally because looking at the doc() method, it takes a path and additional path segments. But updated getpath to just get the full path with slashes & now here, we just pass in path.